python/ql/src/semmle/python/frameworks/Stdlib.qll

```diff
@@ -1209,14 +1209,21 @@ private module Stdlib {
       private class AdditionalTaintStep extends TaintTracking::AdditionalTaintStep {
         override predicate step(DataFlow::Node nodeFrom, DataFlow::Node nodeTo) {
           // Methods
-          nodeFrom = instance() and
-          nodeTo in [getvalueRef(), getfirstRef(), getlistRef()]
-          or
-          nodeFrom = getvalueRef() and nodeTo = getvalueResult()
-          or
-          nodeFrom = getfirstRef() and nodeTo = getfirstResult()
+          exists(DataFlow::AttrRead read | read = nodeTo and read.getObject() = nodeFrom |
+            nodeFrom = instance() and
+            nodeTo in [getvalueRef(), getfirstRef(), getlistRef()]
+          )
```

Comment:
```suggestion
          nodeFrom = nodeTo.(DataFlow::AttrRead).getObject() and
          nodeFrom = instance() and
          nodeTo in [getvalueRef(), getfirstRef(), getlistRef()]
```

Reply:
@tausbn seems to push towards this style, and it is nice to get rid of the exists..

Reply:
Sure. I don't really mind either way :shrug:

Reply:
Cool, then let us go for the compact one, since I am getting used to reading that now (and the elaborate one weakly signals that the compact one is not possible).